pkg/cloud/cloud.go

```diff
@@ -1190,16 +1237,33 @@ func (c *cloud) getInstance(ctx context.Context, nodeID string) (*ec2.Instance,
 		if aws.StringValue(nextToken) == "" {
 			break
 		}
-		request.NextToken = nextToken
+		request.NextToken = nextToken //
 	}
+	return instances, nil
+}
 
-	if l := len(instances); l > 1 {
-		return nil, fmt.Errorf("found %d instances with ID %q", l, nodeID)
-	} else if l < 1 {
-		return nil, ErrNotFound
+func (c *cloud) getInstance(ctx context.Context, nodeID string) (*ec2.Instance, error) {
+	request := &ec2.DescribeInstancesInput{
+		InstanceIds: []*string{&nodeID},
 	}
 
-	return instances[0], nil
+	if c.bm == nil {
+		instances, err := describeInstances(ctx, c.ec2, request)
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO Q should I bring these errors outside of if because they apply to batched and unbatched versions of this?
+		if l := len(instances); l > 1 {
+			return nil, fmt.Errorf("found %d instances with ID %q", l, nodeID)
+		} else if l < 1 {
+			return nil, ErrNotFound
+		}
```

Comment:
The batcher is designed to execute multiple tasks as a batch and then distribute each individual result back to the respective caller - each task in the batch has its own result channel, and each channel is intended to receive exactly one result corresponding to its task.

We could add the extra check for sanity but I believe the code as written is resilient and is consistent with `describeVolumes`, which is battle tested.